src/Annotation/ReadOnlyProperty.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Annotation;
+
+/**
+ * @final
+ *
+ * @Annotation
+ * @Target({"CLASS","PROPERTY"})
+ */
+/* final */ class ReadOnlyProperty
```

Comment:
This is purposefully left as not final and only annotated `@final` to help with B/C and code complexity.  Otherwise, there'd have to be a lot of PHP version conditionals and checks for both annotation classes.  In 4.0, when the deprecated `@ReadOnly` annotation is removed the final should be uncommented.

Reply:
makes sense! i'm ok with it